lib/service/cfg.go

```diff
@@ -302,6 +302,45 @@ func (cfg *Config) DebugDumpToYAML() string {
 	return string(out)
 }
 
+// ComponentsCount returns the number of components enabled.
+func (cfg *Config) ComponentCount() int {
+	componentCount := 0
+
+	if cfg.Auth.Enabled {
+		componentCount++
+	}
+
+	if cfg.SSH.Enabled {
+		componentCount++
+	}
+
+	proxyConfig := cfg.Proxy
+	if proxyConfig.Enabled {
+		componentCount++
+	}
+	if proxyConfig.Kube.Enabled && !proxyConfig.Kube.ListenAddr.IsEmpty() && !proxyConfig.DisableReverseTunnel {
```

Comment:
How does the `proxyConfig.DisableReverseTunnel` impact `kubernetes_service`  ? 

Also I think that there are some cases where `proxyConfig.Kube.ListenAddr` is empty but kube service should be started

For example when teleport is stared as kube agent: 
```yaml
teleport:
    kubernetes_service:
    enabled: true
    kubeconfig_file: /path/to/kubeconfig

    auth_service:
      enabled: false
    
    ssh_service:
      enabled: false
    
    proxy_service:
      enabled: false
    
    app_service:
      enabled: false
    
    db_service:
      enabled: false
  ```

Reply:
> How does the `proxyConfig.DisableReverseTunnel` impact `kubernetes_service` ?

I'm not familiar enough with Teleport to able to answer this.

I came up with `if proxyConfig.Kube.Enabled && !proxyConfig.Kube.ListenAddr.IsEmpty() && !proxyConfig.DisableReverseTunnel {` by putting the following two conditions together:

https://github.com/gravitational/teleport/blob/b70c1d6d63ad2c91de27ec527d15a5ceb9df4886/lib/service/service.go#L3043
https://github.com/gravitational/teleport/blob/b70c1d6d63ad2c91de27ec527d15a5ceb9df4886/lib/service/service.go#L2581-L2588

`ComponentCount` counts the number of components that send heartbeats. When the condition above holds, it seems that [the proxy kube component](https://github.com/gravitational/teleport/blob/b70c1d6d63ad2c91de27ec527d15a5ceb9df4886/lib/service/service.go#L3053) will be started and [will send heartbeats](https://github.com/gravitational/teleport/blob/b70c1d6d63ad2c91de27ec527d15a5ceb9df4886/lib/service/service.go#L3080).

> Also I think that there are some cases where `proxyConfig.Kube.ListenAddr` is empty but kube service should be started
> 
> For example when teleport is stared as kube agent:
> 
> ```yaml
> teleport:
>     kubernetes_service:
>     enabled: true
>     kubeconfig_file: /path/to/kubeconfig
> 
>     auth_service:
>       enabled: false
>     
>     ssh_service:
>       enabled: false
>     
>     proxy_service:
>       enabled: false
>     
>     app_service:
>       enabled: false
>     
>     db_service:
>       enabled: false
> ```

I think that in this case this function will correctly return `1` due to this piece of code below:
```go
	if cfg.Kube.Enabled {
		componentCount++
	}
```